### PR TITLE
Adds fmt string formatting

### PR DESCRIPTION
**Pull request by jamiebramwell**
This adds `fmt` string formatting and tests the build against axom.

### CODE REVIEW

**Comment by jamiebramwell** [comment]
I need to go update the docker containers to include axom. I believe that will fix the azure builds.

File: src/solvers/base_solver.cpp [hunk 1]
```diff
@@ -189,10 +188,8 @@ void BaseSolver::OutputState() const
 
     case OutputType::GLVis: {
       for (auto &state : m_state) {
-        std::ostringstream sol_name;
-        sol_name << m_root_name << "-" << state.name << "." << std::setfill('0') << std::setw(6) << m_cycle << "."
-                 << std::setfill('0') << std::setw(6) << m_rank - 1;
-        std::ofstream osol(sol_name.str().c_str());
+        std::string   sol_name = fmt::format("{0}-{1}.{2:0>6}.{3:0>6}", m_root_name, state.name, m_cycle, m_rank);
```
Comment (samuelpmish) [inline src/solvers/base_solver.cpp:191]: `m_rank-1` became `m_rank`, not sure if intended or not

Comment (jamiebramwell) [reply src/solvers/base_solver.cpp:191]: Yep, that was intended. I'm not sure how this worked at all before... 🤷‍♀️

**Comment by jamiebramwell** [comment]
LGTM

**Comment by jamiebramwell** [comment]
LGTM

**Comment by jamiebramwell** [comment]
Any thoughts on moving to intel 19.0.0 to resolve [this issue](https://github.com/LLNL/axom/issues/237)?

File: host-configs/quartz-toss_3_x86_64_ib-clang@4.0.0.cmake [hunk 2]
```diff
@@ -33,7 +33,7 @@ set(MPI_CXX_COMPILER "/usr/tce/packages/mvapich2/mvapich2-2.2-clang-4.0.0/bin/mp
 #---------------------------------------
 # Library Dependencies
 #---------------------------------------
-set(TPL_ROOT "/usr/WS2/smithdev/libs/toss_3_x86_64_ib/2020_04_16_21_23_26/clang-4.0.0" CACHE PATH "")
+set(TPL_ROOT "/usr/WS2/smithdev/libs/toss_3_x86_64_ib/2020_03_25_19_18_28/clang-4.0.0" CACHE PATH "")
```
Comment (white238) [inline host-configs/quartz-toss_3_x86_64_ib-clang@4.0.0.cmake:36]: Sorry...

Comment (white238) [reply host-configs/quartz-toss_3_x86_64_ib-clang@4.0.0.cmake:36]: The cmake path is right.. but the TPL_ROOT needs to be back to the 4_16

Comment (jamiebramwell) [reply host-configs/quartz-toss_3_x86_64_ib-clang@4.0.0.cmake:36]: Yeah, I'm trying to rebuild the TPLs with mixed success.

**Comment by jamiebramwell** [comment]
LGTM

**Pull request merged**